src/data/finset.lean

```diff
@@ -2553,6 +2553,42 @@ by { rw [finset.disjoint_left, set.disjoint_left], refl }
 
 end disjoint
 
+/--
+Given a set A and a set B inside it, we can shrink A to any appropriate size, and keep B
+inside it.
+-/
+lemma exists_intermediate_set {A B : finset α} (i : ℕ)
+  (h₁ : i + card B ≤ card A) (h₂ : B ⊆ A) :
+  ∃ (C : finset α), B ⊆ C ∧ C ⊆ A ∧ card C = i + card B :=
+begin
+  classical,
+  rcases nat.le.dest h₁ with ⟨k, _⟩,
+  clear h₁,
+  induction k with k ih generalizing A,
+  { exact ⟨A, h₂, subset.refl _, h.symm⟩ },
+  { have : (A \ B).nonempty,
+    { rw [← card_pos, card_sdiff h₂, ← h, nat.add_right_comm,
+          nat.add_sub_cancel, nat.add_succ],
+      apply nat.succ_pos },
+    rcases this with ⟨a, ha⟩,
+    have z : i + card B + k = card (erase A a),
+    { rw [card_erase_of_mem, ← h, nat.add_succ, nat.pred_succ],
+      rw mem_sdiff at ha,
+      exact ha.1 },
+    rcases ih _ z with ⟨B', hB', B'subA', cards⟩,
+    { exact ⟨B', hB', trans B'subA' (erase_subset _ _), cards⟩ },
+    { rintros t th,
+      apply mem_erase_of_ne_of_mem _ (h₂ th),
+      rintro rfl,
+      rw mem_sdiff at ha,
+      tauto } }
```

Comment:
```suggestion
     exact not_mem_sdiff_of_mem_right th ha } }
```
This whole subgoal could be factored out to a standalone theorem in the `erase` section, named something like `subset_erase_of_mem_sdiff` (?).